cmake/modules/FindSwiftc.cmake

```diff
@@ -0,0 +1,16 @@
+# Usage: find_package(Swiftc)
+#
+# If successful the following variables will be defined
+# SWIFTC_FOUND
+# SWIFTC_EXECUTABLE
+
+find_program(SWIFTC_EXECUTABLE
+             NAMES swiftc
+             DOC "Path to 'swiftc' executable")
```

Comment:
We need to check and record the TOOLCHAINS and SDKROOT in use... if not, then what will happen when run as:
```
env TOOLCHAINS=swift cmake ...
```
is we will just find `/usr/bin/swiftc` as the `swiftc`, but if we run again with a different TOOLCHAINS variable it won't work. I'd prefer if the configure step would bake in the value for TOOLCHAINS, I think.